examples/text-generation/run_generation.py

```diff
@@ -211,6 +221,8 @@ def main():
     )
 
     args = parser.parse_args()
+    if args.max_length is None and args.max_new_tokens is None:
+        args.max_new_tokens = 100
```

Comment:
To keep back compatibility. In current case we can not specify `--max_new_tokens` and it would ge a default of 100

Since it was optional I have: length_group = parser.add_mutually_exclusive_group(required=**False**)

However to maintain back-compat if its not specified, I put default=100

Note I cannot put the default value in " length_group.add_argument(", because in the case where `--max_length` is specified and `--max_new_tokens` isnt, arg parser will assign it default of 100, which will result in both `max_length` and `max_new_tokens` getting values
Hence teh default is set here, outside argparser